test/net/sourceforge/kolmafia/textui/command/PrefTraceCommandTest.java

```diff
@@ -0,0 +1,53 @@
+package net.sourceforge.kolmafia.textui.command;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.not;
+import static org.hamcrest.core.StringContains.containsString;
+
+import internal.helpers.Cleanups;
+import internal.helpers.Player;
+import internal.helpers.RequestLoggerOutput;
+import net.sourceforge.kolmafia.KoLCharacter;
+import net.sourceforge.kolmafia.preferences.Preferences;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class PrefTraceCommandTest extends AbstractCommandTestBase {
+
+  public PrefTraceCommandTest() {
+    this.command = "ptrace";
+  }
+
+  @BeforeEach
+  public void setup() {
+    KoLCharacter.reset("ptrace");
+    Preferences.reset("ptrace");
+  }
+
+  @Test
+  public void tracesPreferencesChanged() {
+    execute("_VYKEACompanionType");
+
+    RequestLoggerOutput.startStream();
+    Cleanups cleanups = Player.setProperty("_VYKEACompanionType", "couch");
+    try (cleanups) {
+      var text = RequestLoggerOutput.stopStream();
+      assertThat(text, containsString("ptrace: _VYKEACompanionType = couch"));
+    }
+  }
```

Comment:
I notice that you are leaving the PreferenceListener active, rather than clearing it via execute("").
Intended?

Reply:
It was intended when I wrote it. The tests that clear it do so to test particular code paths.

But I think you're right, I should be clearing up the listeners. As it stands, whichever test runs last will leak its listeners into the remaining tests.